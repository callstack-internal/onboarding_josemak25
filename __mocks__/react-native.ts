jest.mock("react-native", () => {
  const RN = jest.requireActual("react-native"); // use original implementation, which comes with mocks out of the box

  // mock modules/components created by assigning to NativeModules
  RN.NativeModules.SourceCode = {
    scriptURL: "http://localhost:3000/",
  };

  return RN;
});
