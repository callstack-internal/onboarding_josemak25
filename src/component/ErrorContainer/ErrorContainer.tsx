import React, { FC } from "react";
import {
  View,
  Text,
  TextStyle,
  ViewStyle,
  TouchableOpacity,
} from "react-native";

import { makeUseStyles } from "../../helpers/makeUseStyles";

type ErrorContainerProps = {
  message: string;
  onRetry: () => void;
  messageStyle?: TextStyle;
  retryTextStyle?: TextStyle;
  retryButtonContainerStyle?: ViewStyle;
};

export const ErrorContainer: FC<ErrorContainerProps> = ({
  message,
  onRetry,
  messageStyle,
  retryTextStyle,
  retryButtonContainerStyle,
}) => {
  const { styles } = useStyles();

  return (
    <View style={styles.container}>
      <Text style={[styles.message, messageStyle]}>{message}</Text>
      <TouchableOpacity
        onPress={onRetry}
        testID="retry_network_call"
        accessibilityRole="button"
        accessibilityLabel="retry fetching weather"
        style={[styles.retryButtonContainer, retryButtonContainerStyle]}>
        <Text style={[styles.retryText, retryTextStyle]}>Retry</Text>
      </TouchableOpacity>
    </View>
  );
};

const useStyles = makeUseStyles(({ palette, layout, fonts }) => ({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  message: {
    textAlign: "center",
    fontSize: fonts.size.md,
    marginVertical: layout.gutter,
    marginHorizontal: layout.gutter * 2,
  },
  retryButtonContainer: {
    height: 30,
  },
  retryText: {
    color: palette.text,
    fontFamily: fonts.variants.bold,
  },
}));
