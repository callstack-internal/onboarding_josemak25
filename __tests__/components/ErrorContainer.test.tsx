import React from "react";

import { render, fireEvent, screen } from "@testing-library/react-native";
import { ErrorContainer } from "../../src/component/ErrorContainer";

describe("components / ErrorContainer.tsx", () => {
  it("should render correctly", () => {
    const error = "Check your internet";
    const onRetryPress = jest.fn();

    render(<ErrorContainer message={error} onRetry={onRetryPress} />);

    expect(screen.getByText(error)).toBeTruthy();
    expect(screen.getByTestId("retry_network_call")).toBeTruthy();
    fireEvent.press(screen.getByTestId("retry_network_call"));
    expect(onRetryPress).toHaveBeenCalled();
    expect(onRetryPress).toHaveBeenCalledTimes(1);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
