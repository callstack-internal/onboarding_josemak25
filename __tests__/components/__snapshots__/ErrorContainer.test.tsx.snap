// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components / ErrorContainer.tsx should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "marginHorizontal": 24,
          "marginVertical": 12,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Check your internet
  </Text>
  <View
    accessibilityLabel="retry fetching weather"
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 30,
        "opacity": 1,
      }
    }
    testID="retry_network_call"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "manrope-bold",
          },
          undefined,
        ]
      }
    >
      Retry
    </Text>
  </View>
</View>
`;
